### PR TITLE
Bug fixes and cleanups

### PR DESCRIPTION
**Pull request by domesticmouse**
PTAL @samthor && @markmcd

### CODE REVIEW

File: src/test/java/com/google/maps/DirectionsApiTest.java [hunk 1]
```diff
@@ -394,4 +394,29 @@ public void testGeocodedWaypoints() throws Exception {
...
+   * Test {@code local_icon} for Directions in Paris.
+   */
+  @Test
+  public void testLocalIconInParis() throws Exception {
+    DirectionsResult result = DirectionsApi.newRequest(context)
+        .origin("paris metro bibliotheque francois mitterrand")
+        .destination("paris metro pyramides")
+        .mode(TravelMode.TRANSIT)
+        .await();
+    assertNotNull(result);
```
Comment (samthor) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: I know there's not, but I'm really sad there's not a better way to do this.

Well.. actually. You could just...

```java
String localIcon = "";
try {
  localIcon = result.routes[0].legs[0].....
} catch (NullPointerException e) {
  fail("expected transit...");
}
assertEquals(...);
```

Just my 2c, up to you though.

File: src/main/java/com/google/maps/model/Vehicle.java [hunk 2]
```diff
@@ -39,4 +39,9 @@
    * {@code icon} contains the URL for an icon associated with this vehicle type.
    */
   public String icon;
+
+  /**
+   * {@code local_icon} contains the URL for an icon based on the local transport signage.
+   */
+  public String local_icon;
```
Comment (samthor) [inline src/main/java/com/google/maps/model/Vehicle.java]: Shouldn't this be `localIcon`?

File: src/main/java/com/google/maps/model/PlacesSearchResult.java [hunk 3]
```diff
@@ -67,7 +67,7 @@
   /**
    * types contains an array of feature types describing the given result.
    */
-  public String types[];
+  public AddressType types[];
```
Comment (samthor) [inline src/main/java/com/google/maps/model/PlacesSearchResult.java]: It doesn't look like the code which sets up this value has changed? Or does the JSON decoder intelligently do the right thing for string enums here?

File: src/main/java/com/google/maps/model/AddressType.java [hunk 4]
```diff
@@ -206,6 +206,18 @@
   PLACE_OF_WORSHIP("place_of_worship"),
 
   /**
+   * {@code WARD} indicates a specific type of Japanese locality, to facilitate distinction between
+   * multiple locality components within a Japanese address.
+   */
+  WARD("ward"),
```
Comment (samthor) [inline src/main/java/com/google/maps/model/AddressType.java]: 🇯🇵!

File: src/test/java/com/google/maps/DirectionsApiTest.java (hunk 1 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: The long hand approach makes debugging easier. =)

File: src/main/java/com/google/maps/model/PlacesSearchResult.java (hunk 3 above)
Comment (domesticmouse) [inline src/main/java/com/google/maps/model/PlacesSearchResult.java]: The magic responsible for this is https://github.com/googlemaps/google-maps-services-java/blob/master/src/main/java/com/google/maps/internal/SafeEnumAdapter.java#L64

However, I'm having to revert this change because types on place search results aren't AddressTypes. They just look like them sometimes.

File: src/main/java/com/google/maps/model/AddressType.java (hunk 4 above)
Comment (domesticmouse) [inline src/main/java/com/google/maps/model/AddressType.java]: That's less than understandable on my linux desktop. Yay Emoji. =)

**Comment by samthor** [comment]
LGTM

File: src/test/java/com/google/maps/PlacesApiIntegrationTest.java [hunk 5]
```diff
@@ -273,7 +275,7 @@ public void testRadarSearchRequestByKeyword() throws Exception {
         .keyword("pub").await();
     assertNotNull(response);
     assertNotNull(response.results);
-    assertTrue(175 < response.results.length);
+    assertTrue(100 < response.results.length);
```
Comment (markmcd) [inline src/test/java/com/google/maps/PlacesApiIntegrationTest.java]: What happened to all of the pubs in Sydney? :fearful:

File: src/test/java/com/google/maps/DirectionsApiTest.java (hunk 1 above)
Comment (markmcd) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: I wouldn't even catch the NPE. JUnit will catch and report the error.

For debugging purposes would it be cleaner to just log the request/response? Or use a breakpoint?

File: src/test/java/com/google/maps/PlacesApiIntegrationTest.java [hunk 6]
```diff
@@ -176,8 +178,8 @@ public void testPhoto() throws Exception {
     assertNotNull(image);
   }
 
-  @Test
-  public void testPizzaInNewYork() throws Exception {
+    @Test
```
Comment (markmcd) [inline src/test/java/com/google/maps/PlacesApiIntegrationTest.java]: nit: spaces

**Comment by domesticmouse** [comment]
PTAL @broady

File: src/test/java/com/google/maps/PlacesApiIntegrationTest.java (hunk 5 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/PlacesApiIntegrationTest.java]: Late night trading restrictions?

File: src/test/java/com/google/maps/PlacesApiIntegrationTest.java (hunk 6 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/PlacesApiIntegrationTest.java]: Fixed

File: src/test/java/com/google/maps/DirectionsApiTest.java (hunk 1 above)
Comment (domesticmouse) [inline src/test/java/com/google/maps/DirectionsApiTest.java]: Dropping most of the assertNotNull tests.

File: src/main/java/com/google/maps/model/Vehicle.java (hunk 2 above)
Comment (domesticmouse) [inline src/main/java/com/google/maps/model/Vehicle.java]: Fixed
